pennylane_lightning_gpu/src/simulator/StateVectorCudaMPI.hpp

```diff
@@ -941,6 +955,364 @@ class StateVectorCudaMPI
         return expect_val;
     }
 
+    /**
+     * @brief Scatter a CSR (Compress Sparse Row) format matrix.
+     *
+     * @tparam index_type Integer type used as indices of the sparse matrix.
+     * @param matrix CSR (Compress Sparse Row) format matrix.
+     * @param root Root rank of the scatter operation.
+     */
+    template <class index_type>
+    auto scatterCSRMatrix(std::vector<CSRMatrix<Precision, index_type>> &matrix,
+                          size_t root) -> CSRMatrix<Precision, index_type> {
+        // Bcast num_rows and num_cols
+        size_t local_num_rows;
+        size_t num_col_blocks;
+        std::vector<size_t> nnzs;
+
+        local_num_rows = size_t{1} << this->getNumLocalQubits();
+        num_col_blocks = mpi_manager_.getSize();
+
+        if (mpi_manager_.getRank() == root) {
+            for (size_t j = 0; j < matrix.size(); j++) {
+                nnzs.push_back(matrix[j].values.size());
+            }
+        }
+
+        size_t local_nnz = mpi_manager_.scatter<size_t>(nnzs, 0)[0];
+
+        CSRMatrix<Precision, index_type> localCSRMatrix(local_num_rows,
+                                                        local_nnz);
+
+        if (mpi_manager_.getRank() == root) {
+            localCSRMatrix.values = matrix[0].values;
+            localCSRMatrix.csrOffsets = matrix[0].csrOffsets;
+            localCSRMatrix.columns = matrix[0].columns;
```

Comment:
So we only use the matrix values at index 0?
Do we need to use a vector of matrices then?

Reply:
I think that block is to avoid rank-0 to rank-0 MPI calls; values are copied directly from `matrix[0]` to `localCSRMatrix`. The other entries go to rank-1, rank-2, etc.

Reply:
Just as @vincentmr said, this block is to avoid rank0 to rank0 MPI calls.